auto-value-gson/src/test/java/com/ryanharter/auto/value/gson/AutoValueGsonExtensionTest.java

```diff
@@ -178,35 +171,136 @@ public void simple() {
         + "      }\n"
         + "      jsonWriter.beginObject();\n"
         + "      jsonWriter.name(\"a\");\n"
-        + "      string_adapter.write(jsonWriter, object.a());\n"
+        + "      if (object.a() == null) {\n"
+        + "        jsonWriter.nullValue();\n"
+        + "      } else {\n"
+        + "        if (string_adapter == null) {\n"
+        + "          string_adapter = gson.getAdapter(java.lang.String.class);\n"
+        + "        }\n"
+        + "        string_adapter.write(jsonWriter, object.a());\n"
+        + "      }\n"
         + "      jsonWriter.name(\"b\");\n"
-        + "      array__int_adapter.write(jsonWriter, object.b());\n"
+        + "      if (object.b() == null) {\n"
+        + "        jsonWriter.nullValue();\n"
+        + "      } else {\n"
+        + "        if (array__int_adapter == null) {\n"
+        + "          array__int_adapter = gson.getAdapter(int[].class);\n"
+        + "        }\n"
+        + "        array__int_adapter.write(jsonWriter, object.b());\n"
+        + "      }\n"
         + "      jsonWriter.name(\"c\");\n"
+        + "      if (int__adapter == null) {\n"
+        + "        int__adapter = gson.getAdapter(java.lang.Integer.class);\n"
+        + "      }\n"
         + "      int__adapter.write(jsonWriter, object.c());\n"
         + "      jsonWriter.name(\"_D\");\n"
-        + "      string_adapter.write(jsonWriter, object.d());\n"
+        + "      if (object.d() == null) {\n"
+        + "        jsonWriter.nullValue();\n"
+        + "      } else {\n"
+        + "        if (string_adapter == null) {\n"
+        + "          string_adapter = gson.getAdapter(java.lang.String.class);\n"
+        + "        }\n"
+        + "        string_adapter.write(jsonWriter, object.d());\n"
+        + "      }\n"
         + "      jsonWriter.name(\"e\");\n"
-        + "      string_adapter.write(jsonWriter, object.e());\n"
+        + "      if (object.e() == null) {\n"
+        + "        jsonWriter.nullValue();\n"
+        + "      } else {\n"
+        + "        if (string_adapter == null) {\n"
+        + "          string_adapter = gson.getAdapter(java.lang.String.class);\n"
```

Comment:
Why are the FQCNs of these types being used instead of imports here? You're likely calling `toString()` or using `$L` with a `TypeName` instead of using `$T`.

Reply:
I totally agree - FQCNs are unnecessary. Will fix!